product_dimension/models/product.py

```diff
@@ -24,11 +23,7 @@ def onchange_calculate_volume(self):
 
     def convert_to_meters(self, measure, dimensional_uom):
         uom_meters = self.env['product.uom'].search([('name', '=', 'm')])
```

Comment:
Why not use the XML ID ? 
uom_meters = self.env.ref('product.product_uom_meter')